torch/csrc/jit/passes/onnx/unpack_quantized_weights.cpp

```diff
@@ -145,13 +145,24 @@ void unpackQuantizedWeightsHelper(
       throw std::runtime_error(
           "getValues: Quantized weight value not found amongst constant parameters.");
     }
-    at::Tensor packed_weight = itr->second;
-    auto op = Dispatcher::singleton().findSchema({unpack_fn, ""});
-    assert(op.has_value());
-    std::tuple<at::Tensor, c10::optional<at::Tensor>> result = callOpUnboxed<
-        std::tuple<at::Tensor, c10::optional<at::Tensor>>,
-        at::Tensor>(*op, packed_weight);
-    at::Tensor unpacked_weight = std::get<0>(result);
+    at::Tensor unpacked_weight;
+    c10::optional<at::Tensor> bias;
+
+    if (itr->second.isTuple()) {
+      // Pre-unpacked weights. Comes from Linear weights which are
```

Comment:
this is a magical dependency on the LinearWeights format. At least put a comment there that if it changes - you need to update this place too (or make it a static method of LinearWeight class)

Reply:
so why you can't pass the custom class all the way here and call `__getstate__` here? There's a nice symmetry then with the 'else' case which also does unpacking (this is pretty much the same as the comment above)

Reply:
It doesn't fit into the existing infrastructure:
```
======================================================================
ERROR: test_torchbind_lower_graph (__main__.TestScript)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_jit.py", line 4630, in test_torchbind_lower_graph
    onnx_model = export_to_onnx(model, x, input_names)
  File "test/test_jit.py", line 4624, in export_to_onnx
    operator_export_type=torch.onnx.OperatorExportTypes.ONNX_ATEN_FALLBACK)
  File "/data/users/jamesreed/pytorch/torch/onnx/__init__.py", line 172, in export
    custom_opsets, enable_onnx_checker, use_external_data_format)
  File "/data/users/jamesreed/pytorch/torch/onnx/utils.py", line 92, in export
    use_external_data_format=use_external_data_format)
  File "/data/users/jamesreed/pytorch/torch/onnx/utils.py", line 519, in _export
    fixed_batch_size=fixed_batch_size)
  File "/data/users/jamesreed/pytorch/torch/onnx/utils.py", line 340, in _model_to_graph
    in_vars, in_desc = torch.jit._flatten(tuple(args) + tuple(params))
RuntimeError: Only tuples, lists and Variables supported as JIT inputs/outputs. Dictionaries and strings are also accepted but their usage is not recommended. But got unsupported type torch._C.ScriptObject
```

And I don't want to rabbit-hole down another fractal of redesign when this solution just works